homeassistant/components/trace/__init__.py

```diff
@@ -239,7 +241,7 @@ def __init__(
             trace_set_child_id(self.key, self.run_id)
         trace_id_set((self.key, self.run_id))
 
-    def set_trace(self, trace: dict[str, deque[TraceElement]]) -> None:
+    def set_trace(self, trace: dict[str, deque[TraceElement]] | None) -> None:
```

Comment:
I've enabled `None` here for `trace`
Since `self._trace` is already set to be optional, I think this should be OK and it helps with #78368